packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/workflow/Workflow.java

```diff
@@ -20,20 +31,51 @@
 @Data
 @Accessors(chain = true)
 @TSModel
-public class Workflow extends TerariumAsset implements Serializable {
+@Entity
+public class Workflow extends TerariumAsset {
 
 	@Serial
 	private static final long serialVersionUID = -1565930053830366145L;
 
-
+	@Schema(defaultValue = "My New Workflow")
 	private String name;
 
 	private String description;
 
 	private Transform transform;
 
-	private Object nodes;
+	@Convert(converter = JsonConverter.class)
+	@JdbcTypeCode(SqlTypes.JSON)
+	private List<WorkflowNode> nodes;
+
+	@Convert(converter = JsonConverter.class)
+	@JdbcTypeCode(SqlTypes.JSON)
+	private List<WorkflowEdge> edges;
+
+	public Workflow clone() {
+		final Workflow clone = new Workflow();
+		clone.setId(UUID.randomUUID());
+		clone.setName(this.getName());
+		clone.setDescription(this.getDescription());
+		clone.setTransform(this.getTransform());
+
+		final Map<UUID, WorkflowNode> oldToNew = new HashMap<>();
+
+		clone.setNodes(new ArrayList<>());
+		for (final WorkflowNode node : nodes) {
+			final WorkflowNode clonedNode = node.clone();
+			oldToNew.put(node.getId(), clonedNode);
+			clone.getNodes().add(clonedNode);
+		}
 
-	private Object edges;
+		clone.setEdges(new ArrayList<>());
+		for (final WorkflowEdge edge : edges) {
+			final WorkflowEdge clonedEdge = edge.clone();
+			clonedEdge.setSource(oldToNew.get(edge.getSource().getId()));
+			clonedEdge.setTarget(oldToNew.get(edge.getTarget().getId()));
```

Comment:
The source/target assignments look suspicious, if we are cloning it I would expect the source/target nodes to be new and distinct instances? Or do we have a different meaning of cloning here?

Reply:
The UI has a branching function that acts like a clone for a subgraph of the workflow-graph:

https://github.com/DARPA-ASKEM/terarium/blob/main/packages/client/hmi-client/src/services/workflow.ts#L445-L549

Reply:
We are cloning the edge, and then setting the nodes to the newly cloned nodes, but using a lookup from prev id to cloned id with the `oldToNew` map which is `old_node_id -> new_cloned_node`

Reply:
If I am missing things that also need to be cloned, then I need to fix that